lib/src/native/realm_core.dart

```diff
@@ -450,6 +453,63 @@ class _RealmCore {
     return RealmHandle._(realmPtr);
   }
 
+  RealmSchema readSchema(Realm realm) {
+    return using((Arena arena) {
+      final classKeys = _getValues<Uint32, int>(
```

Comment:
This requires a lot of code duplication as we'll need to inline the _getValues function everywhere it's used. I don't see how this would be helpful or make the codebase more maintainable.

Reply:
I have strong opinion against writing such code. We can discuss how to improve it but this kind of code is totally degrading the codebase. I would suggest we avoid late generalizations like these.

Reply:
I have inlined the calls to _getValues. I still believe it's an unnecessary amount of code duplication, but don't care enough to hold this PR captive to resolve it.